mysticeti-core/src/statement_block_validator.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+
+use crate::types::StatementBlock;
+
+/// The interfaces to verify the legitimacy of a statement block's contents.
+#[async_trait]
+pub trait StatementBlockValidator: Clone + Send + Sync + 'static {
```

Comment:
Not: i wonder if something like BlockFilter is a better/shorter name (up to you, just a suggestion)

Reply:
I leave it as `StatementBlockValidator`: `BlockFilter` gives me a feeling that it filters partial invalid contents in a block and keep good ones, which does not reflect what it does in practice

Reply:
but we can call it `BlockValidator` to make it shorter, wdyt?

Reply:
Yeah no strong opinion on the name.
Also just noticed that we have :Clone requirement here, but I don't think we actually need it since it's wrapped with Arc anyway?

Reply:
good catch, updated both in  https://github.com/MystenLabs/mysticeti/pull/33